src/locale/localed.c

```diff
@@ -661,6 +661,11 @@ static int method_set_x11_keyboard(sd_bus_message *m, void *userdata, sd_bus_err
                 log_error_errno(r, "Failed to set X11 keyboard layout: %m");
                 return sd_bus_error_set_errnof(error, r, "Failed to set X11 keyboard layout: %m");
         }
+        r = vconsole_write_data(c);
```

Comment:
This should be earlier than `x11_write_data()`, as now vconsole.conf has precedence over X configuration.

Also, `x11_convert_to_vconsole_and_emit()` below also writes vconsole.conf. Please try to convert earlier, and write vconsole.conf only once.

Reply:
so 
```
diff --git a/src/locale/localed.c b/src/locale/localed.c
index 209d36a0ed..132a61ddae 100644
--- a/src/locale/localed.c
+++ b/src/locale/localed.c
@@ -656,16 +656,16 @@ static int method_set_x11_keyboard(sd_bus_message *m, void *userdata, sd_bus_err
             free_and_strdup(&c->x11_options, options) < 0)
                 return -ENOMEM;
 
-        r = x11_write_data(c);
-        if (r < 0) {
-                log_error_errno(r, "Failed to set X11 keyboard layout: %m");
-                return sd_bus_error_set_errnof(error, r, "Failed to set X11 keyboard layout: %m");
-        }
         r = vconsole_write_data(c);
         if (r < 0) {
                 log_error_errno(r, "Failed to save X11 keyboard layout to /etc/vconsole.conf: %m");
                 return sd_bus_error_set_errnof(error, r, "Failed to save X11 keyboard layout to /etc/vconsole.conf: %m");
         }
+        r = x11_write_data(c);
+        if (r < 0) {
+                log_error_errno(r, "Failed to set X11 keyboard layout: %m");
+                return sd_bus_error_set_errnof(error, r, "Failed to set X11 keyboard layout: %m");
+        }
 
         log_info("Changed X11 keyboard layout to '%s' model '%s' variant '%s' options '%s'",
                  strempty(c->x11_layout),

```
?